### PR TITLE
Ee 19133 add new endpoints audit client

### PR DESCRIPTION
**Pull request by owainrobertsbjss**
Adding methods to access the two new routes in the audit-service, /correlationIds and /historyByCorrelationId. 

As these methods aren't yet used I intend to merge into master once approved.

### CODE REVIEW

File: src/main/java/uk/gov/digital/ho/proving/income/audit/AuditClient.java [hunk 1]
```diff
@@ -51,15 +53,19 @@
...
-                       @Value("${audit.history.endpoint}") String auditHistoryEndpoint,
-                       @Value("${audit.archive.endpoint}") String auditArchiveEndpoint,
-                       @Value("${audit.archive.history.pagesize}") int historyPagesize,
-                       ObjectMapper mapper) {
+                @Value("${audit.history.endpoint}") String auditHistoryEndpoint,
+                @Value("${audit.history.correlationids.endpoint}") String correlationIdsEndpoint,
+                @Value("${audit.history.bycorrelationid.endpoint}") String historyByCorrelationIdEndpoint,
+                @Value("${audit.archive.endpoint}") String auditArchiveEndpoint,
+                @Value("${audit.archive.history.pagesize}") int historyPagesize,
+                ObjectMapper mapper) {
```
Comment (mikehalmaho) [inline src/main/java/uk/gov/digital/ho/proving/income/audit/AuditClient.java]: Is it time for a separate class to hold these properties?  10 constructor args should be breaking checkstyle.

File: src/main/java/uk/gov/digital/ho/proving/income/audit/AuditClient.java [hunk 2]
```diff
@@ -51,15 +53,19 @@
                 RestTemplate restTemplate,
                 RequestData requestData,
                 @Value("${pttg.audit.endpoint}") String auditEndpoint,
```
Comment (Tim-Hammonds) [inline src/main/java/uk/gov/digital/ho/proving/income/audit/AuditClient.java]: The arg list is getting long. Probably worth thinking about how to address this.

**Review by Tim-Hammonds — CHANGES_REQUESTED** [review]

**Review by mikehalmaho — APPROVED** [review]

File: src/main/java/uk/gov/digital/ho/proving/income/audit/AuditClient.java (hunk 2 above)
Comment (owainrobertsbjss) [reply src/main/java/uk/gov/digital/ho/proving/income/audit/AuditClient.java]: Agreed - I'll do that now

File: src/main/java/uk/gov/digital/ho/proving/income/audit/AuditClient.java (hunk 1 above)
Comment (owainrobertsbjss) [reply src/main/java/uk/gov/digital/ho/proving/income/audit/AuditClient.java]: Yes it probably is so I will do that now. Interestingly it seems we don't have checkstyle or similar in this project.

**Review by mikehalmaho — APPROVED** [review]

**Review by Tim-Hammonds — APPROVED** [review]

**Pull request merged**